adam-core/src/main/scala/org/bdgenomics/adam/hbase/HBaseFunctions.scala

```diff
@@ -0,0 +1,587 @@
+    table.put(put)
+  }
+
+  /**
+   * loads Sequence Dictionary data from HBase
+   *
+   * @param dao
+   * @param HbaseTableName
+   * @param sequenceDictionaryId
+   * @return
+   */
+
+  private[hbase] def loadSequenceDictionaryFromHBase(dao: HBaseSparkDAO,
+                                                     HbaseTableName: String,
+                                                     sequenceDictionaryId: String): SequenceDictionary = {
+
+    val table = dao.getTable(TableName.valueOf(HbaseTableName))
+
+    val myGet = new Get(Bytes.toBytes(sequenceDictionaryId))
+    val result = table.get(myGet)
+    val dataBytes = result.getValue(Bytes.toBytes("meta"), Bytes.toBytes("contig"))
+
+    val contigDatumReader: DatumReader[Contig] =
+      new SpecificDatumReader[Contig](scala.reflect.classTag[Contig]
+        .runtimeClass
+        .asInstanceOf[Class[Contig]])
+
+    val decoder = DecoderFactory.get().binaryDecoder(dataBytes, null)
+    var resultList = new ListBuffer[Contig]
+    while (!decoder.isEnd) {
+      resultList += contigDatumReader.read(null, decoder)
+    }
+
+    SequenceDictionary.fromAvro(resultList)
+
+  }
+
+  /**
+   * save Sample Metadata to HBase
+   *
+   * @param dao
+   * @param hbaseTableName
+   * @param samples
+   */
+
+  private[hbase] def saveSampleMetadataToHBase(dao: HBaseSparkDAO,
+                                               hbaseTableName: String,
+                                               samples: Seq[Sample]): Unit = {
+
+    val table: Table = dao.getTable(TableName.valueOf(hbaseTableName))
+
+    val baos: java.io.ByteArrayOutputStream = new ByteArrayOutputStream()
+    val encoder = EncoderFactory.get().binaryEncoder(baos, null)
+    val sampleDatumWriter: DatumWriter[Sample] =
+      new SpecificDatumWriter[Sample](scala.reflect.classTag[Sample].runtimeClass.asInstanceOf[Class[Sample]])
+    samples.foreach((x) => {
+      sampleDatumWriter.write(x, encoder)
+      encoder.flush()
+      baos.flush()
+
+      val curr_sampleid = x.getSampleId
+
+      val put = new Put(Bytes.toBytes(curr_sampleid))
+      put.addColumn(Bytes.toBytes("meta"), Bytes.toBytes("sampledata"), baos.toByteArray)
+      table.put(put)
+
+    })
+  }
+
+  /**
+   * load sample Metadata from HBase
+   *
+   * @param dao
+   * @param hbaseTableName
+   * @param sampleIds
+   * @return
+   */
+  private[hbase] def loadSampleMetadataFromHBase(dao: HBaseSparkDAO,
+                                                 hbaseTableName: String,
+                                                 sampleIds: Seq[String]): Seq[Sample] = {
+
+    val table = dao.getTable(TableName.valueOf(hbaseTableName))
+
+    var resultList = new ListBuffer[Sample]
+
+    sampleIds.foreach((sampleId) => {
+      val get = new Get(Bytes.toBytes(sampleId))
+      val result = table.get(get)
+      val dataBytes = result.getValue(Bytes.toBytes("meta"), Bytes.toBytes("sampledata"))
+
+      val sampleDatumReader: DatumReader[Sample] =
```

Comment:
You should be able to create the `DatumReader[Sample]` outside of the loop and reuse it, right?